tests/console/puppet.pm

```diff
@@ -14,12 +14,17 @@ use base "consoletest";
 use strict;
 use testapi;
 use utils;
+use version_utils 'is_sle';
 
 sub run {
     select_console 'root-console';
     pkcon_quit;
-    my $output      = "puppet cert list --all | grep -woh puppetslave.local > /dev/$serialdev";
-    my $puppet_conf = <<"EOF";
+    if (script_run('zypper se puppet') == 104 && is_sle('15+')) {
+        record_soft_failure 'bsc#1092498 - puppet disappeared from packagehub or was never added';
```

Comment:
better use an early return here and leave out the else below, i.e.

```perl
if (script_run('zypper se puppet') == 104 && is_sle('15+')) {
    return record_soft_failure …
}
…
```

Reply:
exactly, that is the style of code that I was looking for from the beginning. Thanks.